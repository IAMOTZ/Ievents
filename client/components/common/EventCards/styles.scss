@import "../../../styles/variables";

.event-card {
  border-color: $custom-blue;
  .card-header {
    border-bottom-color: $custom-blue;
  }
  .card-body {
    .description {
      height: 75px;
    }
    .read-more-link {
      font-size: 13px;
    }
    .date-and-status {
      font-size: 15px;
      display: flex;
      justify-content: space-between;
    }
  }
  .card-footer {
    border-top-color: $custom-blue;
    .action-links {
      display: flex;
      justify-content: space-between;
      a, span {
        color: #343a40;
        font-size: 15px;
      }
    }
  }
}
