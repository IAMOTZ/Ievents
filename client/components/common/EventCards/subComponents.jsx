import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

/**
 * An helper function to truncate a text.
 * @param {String} text The text to truncate.
 * @returns {String} The truncated text.
 */
const truncateText = (text) => {
  let newText;
  if (text && text.length > 100) {
    newText = text.substr(0, 70);
    newText += '...';
  }
  return newText;
};

/* eslint-disable react/forbid-prop-types */
export const RenderEventDescription = (props) => {
  const { event } = props;
  const truncatedText = truncateText(event.description);
  return (
    <div className="description">
      {
        truncatedText ?
          <span>
            <span>{truncatedText}</span>
            <span className="read-more-link">
              &nbsp;
              <a
                className="no-outline-btn"
                data-toggle="modal"
                href="#event-details-modal"
                onClick={() => props.createModalContent(event)}
                role="button"
              >Read more
              </a>
            </span>
          </span> :
          <span>{event.description}</span>
      }
    </div>
  );
};
RenderEventDescription.propTypes = {
  createModalContent: PropTypes.func.isRequired,
  event: PropTypes.object.isRequired,
};

export const RenderEventFooter = (props) => {
  const { event } = props;
  return (
    <div className="card-footer">
      <div className="action-links">
        <span>
          {
            event.status === 'allowed' ?
              <Link to={`/events/${event.id}/edit`} onClick={props.edit} id={event.id} >
                <i className="fa fa-pencil fa-fw text-dark" />
                <span>Edit Event</span>
              </Link> :
              <span>
                {
                  event.status === 'canceled' ?
                    <i className="fa fa-times text-danger" /> :
                    <i className="fa fa-check text-success" />
                }
                &nbsp;<span className="text-capitalize">{event.status}</span>
              </span>
          }
        </span>
        <span>
          <a href="#confirmation-modal" data-toggle="modal" onClick={props.startDelete} id={event.id}>
            <i className="fa fa-trash text-dark" />&nbsp;
            <span>Delete Event</span>
          </a>
        </span>
      </div>
    </div>
  );
};
RenderEventFooter.propTypes = {
  edit: PropTypes.func.isRequired,
  event: PropTypes.object.isRequired,
  startDelete: PropTypes.func.isRequired,
};

export const RenderEventBody = (props) => {
  const { event } = props;
  return (
    <div className="card-body">
      <div className={event.status === 'allowed' ? '' : 'text-muted'}>
        <RenderEventDescription
          event={event}
          createModalContent={props.createModalContent}
        />
        <p className="mb-1">
          <i className="fa fa-bank fa-fw" />&nbsp;
          <span className="text-capitalize">{event.centerName}</span>
        </p>
        <div className="date-and-status">
          <span>{event.date.replace(/-/g, '/')}&nbsp;</span>
        </div>
      </div>
    </div>
  );
};
RenderEventBody.propTypes = {
  event: PropTypes.object.isRequired,
  createModalContent: PropTypes.func.isRequired,
};

export const RenderEventBodyInTransactions = (props) => {
  const { event } = props;
  return (
    <div className="card-body">
      <RenderEventDescription
        event={event}
        createModalContent={props.createModalContent}
      />
      <p className="mb-1">
        <i className="fa fa-user fa-fw" />&nbsp;
        <span>{event.user.email}</span>
      </p>
      <div className="date-and-status">
        <span>{event.date.replace(/-/g, '/')}&nbsp;</span>
      </div>
      <button
        onClick={props.startEventCancel}
        data-toggle="modal"
        data-target="#confirmation-modal"
        id={event.id}
        className="btn ie-red-button mt-2 btn-block"
      >
        {
          props.cancelingTransactionStarted ?
            <span>Canceling&nbsp;<i className="fa fa-spinner fa-pulse" /></span> : <span>Cancel Event</span>
        }
      </button>
    </div>
  );
};
RenderEventBodyInTransactions.propTypes = {
  createModalContent: PropTypes.func.isRequired,
  event: PropTypes.object.isRequired,
  startEventCancel: PropTypes.func.isRequired,
  cancelingTransactionStarted: PropTypes.bool.isRequired,
};
