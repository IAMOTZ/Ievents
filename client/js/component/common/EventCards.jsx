import React from 'react';

export default (props) => {
  const eventCards = props.events.map((event => {
    return (
      <div className="card text-dark bg-white mb-3 mx-auto" style={{ maxWidth: 20 + 'rem' }} key={event.id}>
        <div className="card-header h4">
          {event.title}
          <a href="" className="ml-4 mr-2">
            <i className="fa fa-pencil"></i>
          </a>
          <a href="#" className="text-primary">
            <i className="fa fa-trash" onClick={props.startDelete} id={event.id}></i>
          </a>
        </div>
        <div className="card-body">
          <p className="card-text">{event.description}</p>
          <span className="text-muted">{event.date}&nbsp;</span>
          <a href="#" className="btn btn-block bg-dark text-white mt-2" onClick={props.remove} id={event.id}>Done</a>
        </div>
      </div>
    );
  }));
  return eventCards;

}