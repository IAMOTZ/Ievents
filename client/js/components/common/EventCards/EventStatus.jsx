import React from 'react';
import PropTypes from 'prop-types';

const EventStatus = (props) => {
  switch (props.status) {
    case ('canceled'): {
      return (
        <div className="card-footer text-danger text-center">
          <i className="fa fa-times-circle" />&nbsp;
          <span>Canceled</span>
        </div>
      );
    }
    case ('done'): {
      return (
        <div className="card-footer text-info text-center">
          <i className="fa fa-exclamation" />&nbsp;
          <span>Done</span>
        </div>
      );
    }
    default: {
      return (
        <div className="card-footer text-success text-center">
          <i className="fa fa-check" />&nbsp;
          <span>Allowed</span>
        </div>
      );
    }
  }
};

EventStatus.defaultProps = {
  status: null,
};
EventStatus.propTypes = {
  status: PropTypes.string,
};
export default EventStatus;
