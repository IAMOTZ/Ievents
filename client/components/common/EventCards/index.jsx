import React from 'react';
import PropTypes from 'prop-types';
import './styles.scss';
import {
  RenderEventBody,
  RenderEventBodyInTransactions,
  RenderEventFooter
} from './subComponents';

/* eslint-disable react/forbid-prop-types */
const EventCards = props => props.events.map(event => (
  <div className="col-lg-4 col-md-6 col-sm-1" key={event.id}>
    <div
      style={{ maxWidth: `${20}rem` }}
      className="card text-dark bg-white mb-3 mx-auto event-card"
    >
      <div className="card-header">
        <span className="h4 mb-0">{event.title}</span>
      </div>
      {
        props.isTransactionsPage ?
          <RenderEventBodyInTransactions
            event={event}
            createModalContent={props.createModalContent}
            startEventCancel={props.startEventCancel}
            cancelingTransactionStarted={props.cancelingTransactionStarted}
          /> :
          <RenderEventBody
            event={event}
            createModalContent={props.createModalContent}
          />
      }
      {
        props.isTransactionsPage ?
          null :
          <RenderEventFooter
            event={event}
            startDelete={props.startDelete}
            edit={props.edit}
          />
      }
    </div>
  </div>
));
EventCards.propTypes = {
  events: PropTypes.array,
};
EventCards.defaultProps = {
  events: []
};


export default EventCards;
