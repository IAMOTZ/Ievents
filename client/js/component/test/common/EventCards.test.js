import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import EventCards from '../../common/EventCards.jsx';


const props = {
  events: [
    {
      id: 1,
      centerId: 1,
      userId: 1,
      title: 'test-event',
      description: 'test description',
      date: '12/34/2020',
      status: 'allowed',
    },
  ],
  centers: [
    { id: 1, name: 'test-center1' },
    { id: 2, name: 'test-center2' },
    { id: 3, name: 'test-center3' },
  ],
  edit: () => { },
  startDelete: () => { },
};

const alterProps = newProps => ({
  events: [...newProps.events],
  centers: props.centers,
  edit: () => { },
  startDelete: () => { },
});

describe('<EventCards />', () => {
  it('should render correcly when status is allowed', () => {
    const tree = renderer.create(
      <Router>
        <EventCards {...props} />
      </Router >
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render correctly when status is canceled', () => {
    const newProps = alterProps({
      events: [
        { ...props.events[0], status: 'canceled' },
      ],
    });
    const tree = renderer.create(
      <Router>
        <EventCards {...newProps} />
      </Router >
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
