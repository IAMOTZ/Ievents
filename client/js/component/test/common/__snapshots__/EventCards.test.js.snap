// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventCards /> should render correcly when status is allowed 1`] = `
<div
  className="card text-dark bg-white mb-3 mx-auto"
  style={
    Object {
      "maxWidth": "20rem",
    }
  }
>
  <div
    className="card-header d-flex justify-content-between"
  >
    <span
      className="h4 mb-0"
    >
      test-event
    </span>
    <span>
      <a
        className="mr-2"
        href="/editEvent"
        onClick={[Function]}
      >
        <i
          className="fa fa-pencil"
          id={1}
          onClick={[Function]}
        />
      </a>
      <a
        className="text-primary"
        href="#"
      >
        <i
          className="fa fa-trash"
          id={1}
          onClick={[Function]}
        />
      </a>
    </span>
  </div>
  <div
    className="card-body"
  >
    <div
      className=""
    >
      <p
        className="card-text"
      >
        test description
      </p>
      <p
        className="mb-1"
      >
        <i
          aria-hidden="true"
          className="fa fa-map-marker fa-fw"
        />
         
        test-center1
      </p>
      <span>
        12/34/2020
         
      </span>
    </div>
  </div>
  <div
    className="card-footer text-success text-center"
  >
    <i
      className="fa fa-check"
    />
     
    <span>
      Allowed
    </span>
  </div>
</div>
`;

exports[`<EventCards /> should render correctly when status is canceled 1`] = `
<div
  className="card text-dark bg-white mb-3 mx-auto"
  style={
    Object {
      "maxWidth": "20rem",
    }
  }
>
  <div
    className="card-header d-flex justify-content-between"
  >
    <span
      className="h4 mb-0 text-muted"
    >
      test-event
    </span>
    <span>
      <a
        className="text-primary"
        href="#"
      >
        <i
          className="fa fa-trash"
          id={1}
          onClick={[Function]}
        />
      </a>
    </span>
  </div>
  <div
    className="card-body"
  >
    <div
      className="text-muted"
    >
      <p
        className="card-text"
      >
        test description
      </p>
      <p
        className="mb-1"
      >
        <i
          aria-hidden="true"
          className="fa fa-map-marker fa-fw"
        />
         
        test-center1
      </p>
      <span>
        12/34/2020
         
      </span>
    </div>
  </div>
  <div
    className="card-footer text-danger text-center"
  >
    <i
      className="fa fa-times-circle"
    />
     
    <span>
      Canceled
    </span>
  </div>
</div>
`;
